django/cantusdb_project/main_app/admin.py

```diff
@@ -1,22 +1,60 @@
+from lib2to3.pytree import Base
 from django.contrib import admin
 from main_app.models import *
 from django.contrib.auth import get_user_model
 # Register your models here.
 
+class BaseModelAdmin(admin.ModelAdmin):
+    exclude = ('created_by', 'last_updated_by')
+
+    def save_model(self, request, obj, form, change):
+        if change:
+            obj.last_updated_by = request.user
+        else:
+            obj.created_by = request.user
+        super().save_model(request, obj, form, change)
+
+class ChantAdmin(BaseModelAdmin):
+    pass
+
+class FeastAdmin(BaseModelAdmin):
+    pass
+
+class GenreAdmin(BaseModelAdmin):
+    pass
+
+class IndexerAdmin(BaseModelAdmin):
+    pass
+
+class NotationAdmin(BaseModelAdmin):
+    pass
+
+class OfficeAdmin(BaseModelAdmin):
+    pass
+
+class ProvenanceAdmin(BaseModelAdmin):
+    pass
+
+class SegmentAdmin(BaseModelAdmin):
+    pass
+
+class SequenceAdmin(BaseModelAdmin):
+    pass
+
 class SourcesUserCanEditInline(admin.TabularInline):
     model = get_user_model().sources_user_can_edit.through
 
-class SourceAdmin(admin.ModelAdmin):
+class SourceAdmin(BaseModelAdmin):
```

Comment:
Why is `SourceAdmin` using `BaseModelAdmin` while other models are using their custom ones? More inline comments will be very helpful.

Reply:
I'm not quite sure I understand. Currently, all of the other models' admins, including `SourceAdmin`, are inheriting from `BaseModelAdmin`.